test/SlowTests/Issues/RavenDB-10607.cs

```diff
@@ -0,0 +1,102 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using FastTests;
+using Xunit;
+
+namespace SlowTests.Issues
+{
+    public class RavenDB_10607 : RavenTestBase
+    {
+        private class Location
+        {
+            public string Id { get; set; }
+            public string Name { get; set; }
+            public List<string> LocationParents { get; set; }
+        }
+
+        private class Result
+        {
+            public string Id { get; set; }
+            public string Name { get; set; }
+            public int? Depth { get; set; }
+        }
+
+        [Fact]
+        public void CanUseCount1()
+        {
+            using (var store = GetDocumentStore())
+            {
+                using (var session = store.OpenSession())
+                {
+                    session.Store(new Location
+                    {
+                        LocationParents = new List<string>
+                        {
+                            "locations/10", "locations/20"
+                        }
+                    });
+
+                    session.SaveChanges();
+                }
+
+                using (var session = store.OpenSession())
+                {
+                    var query = from location in session.Query<Location>()
+                                select new Result
+                                {
+                                    Id = location.Id,
+                                    Depth = location.LocationParents.Count, // Is a list
```

Comment:
What happens if you also add ` Depth2 = location.LocationParents.Count(),` here?